samples/ChangefeedMigrationSample/Program.cs

```diff
@@ -0,0 +1,222 @@
+            // with reserved through pul which has pricing implications. For details
+            // visit: https://azure.microsoft.com/en-us/pricing/details/cosmos-db/
+            await client.CreateDocumentCollectionIfNotExistsAsync(
+                UriFactory.CreateDatabaseUri(this.dbName),
+                new DocumentCollection { Id = this.monitoredCollectionName },
+                new RequestOptions { OfferThroughput = 400 });
+
+            // create lease collect if it does not exist
+            // WARNING: CreateDocumentCollectionIfNotExistsAsync will create a new 
+            // with reserved through pul which has pricing implications. For details
+            // visit: https://azure.microsoft.com/en-us/pricing/details/cosmos-db/
+            await client.CreateDocumentCollectionIfNotExistsAsync(
+            UriFactory.CreateDatabaseUri(this.dbName),
+            new DocumentCollection { Id = this.leaseCollectionName },
+            new RequestOptions { OfferThroughput = 400 });
+            await StartChangeFeedHost();
+        }
+
+        async Task StartChangeFeedHost()
+        {
+            string hostName = Guid.NewGuid().ToString();
+
+            // monitored collection info 
+            DocumentCollectionInfo documentCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri(this.uri),
+                MasterKey = this.secretKey,
+                DatabaseName = this.dbName,
+                CollectionName = this.monitoredCollectionName
+            };
+
+            // lease collection info 
+            DocumentCollectionInfo leaseCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri(this.uri),
+                MasterKey = this.secretKey,
+                DatabaseName = this.dbName,
+                CollectionName = this.leaseCollectionName
+            };
+
+            ChangeFeedEventHost host = new ChangeFeedEventHost(hostName, documentCollectionLocation, leaseCollectionLocation);
+            await host.RegisterObserverAsync<DocumentFeedObserver>();
+            Console.WriteLine("Main program: press Enter to stop...");
+            Console.ReadLine();
+            await host.UnregisterObserversAsync();
+        }
+
+        private async Task UpdateDb()
+        {
+            // Use this function to update data in monitored collection in seperate thread
+            // Returns all documents in the collection.
+            Console.WriteLine("Connect client");
+            DocumentClient client = new DocumentClient(new Uri(this.uri), this.secretKey);
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(this.dbName, this.monitoredCollectionName);
+
+            Console.WriteLine("Connect database");
+            try
+            {
+                Database database = await client.ReadDatabaseAsync(UriFactory.CreateDatabaseUri(this.dbName));
+            }
+            catch (Exception e)
```

Comment:
Why do we catch all exception and still continue to create docs in case there was an exception during connect?